components/post_edit_history/restore_post_modal.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo} from 'react';
+import {defineMessages, useIntl} from 'react-intl';
+
+import {t} from 'utils/i18n';
+
+import {Post} from '@mattermost/types/posts';
+import GenericModal from 'components/generic_modal';
+
+type Props = {
+    post: Post;
+    actions: {
+        handleRestore: (post: Post) => void;
+    };
+    onExited: () => void;
+}
+
+const RestorePostModal = ({post, actions, onExited}: Props) => {
+    const {formatMessage} = useIntl();
+    const onHide = () => onExited();
+
+    const handleRestore = async () => {
+        await actions.handleRestore(post);
+        onHide();
+    };
+
+    const modalMessages = defineMessages({
+        title: {
+            id: t('restore_post.title'),
+            defaultMessage: 'Restore this message?',
+        },
+    });
+
```

Comment:
Please move outside/above the `RestorePostModal` function above the Props Type definition.